docs/user_guide/datasets.rst

```diff
@@ -0,0 +1,333 @@
+  * Show users how bias manifests in the data and in downstream modelling tasks
+  * Suggest best practices for dealing with the Boston Housing data and 
+  alternative benchmarking datasets
+
+.. _dataset_origin:
+
+Dataset Origin and Use
+^^^^^^^^^^^^^^^^^^^^^
+
+Contrary to expectation, the Boston Housing dataset was not developed for 
+economic purposes. `Harrison and Rubenfield (1978) <https://deepblue.lib.umich.edu/bitstream/handle/2027.42/22636/0000186.pdf?sequence=1&isAllowed=y>` 
+developed the dataset to illustrate the issues with using housing market data 
+to measure consumer willingness to pay for clean air. The authors use a 
+`hedonic pricing <https://www.investopedia.com/terms/h/hedonicpricing.asp>` 
+approach, which assumes that the price of a good or service can be modeled as a 
+function of features both internal and external to the good or service. The 
+input to this model was a dataset comprising the Boston Standard Metropolitan 
+Statistical Area, with the *NOX* variable serving as a proxy for air quality. 
+The paper sought to estimate the median value of owner-occupied homes (now 
+*MEDV*), and included the remaining variables to measure other neighborhood 
+characteristics. Further, the authors took the derivative of their housing 
+value equation with respect to *NOX* to measure the "amount of money households
+were willing to pay with respect to air polution levels in their census 
+tracts". The variables in the dataset are representative of the early 1970s 
+and come from a mixture of surveys, administrative records, and other research
+papers. While the paper does define each variable and suggest its impact on 
+the housing value equation, it lacks reasoning for including that particular
+set of variables.
+
+Modern machine learning practicioners have used the Boston Housing dataset as 
+a benchmark to assess the performance of emerging supervised learning 
+techniques. It's featured in `scipy lectures <https://scipy-lectures.org/packages/scikit-learn/auto_examples/plot_boston_prediction.html>`, 
+indexed in the `University of California-Irvine Machine Learning Repository 
+<https://archive.ics.uci.edu/ml/machine-learning-databases/housing/>` and in 
+Carnegie Mellon University's `StatLib <http://lib.stat.cmu.edu/datasets/boston>`, 
+and for a time was included as one of `scikit-learn's <https://scikit-learn.org/stable/modules/generated/sklearn.datasets.load_boston.html>`
+ and tensorflow's standard toy datasets. It has also been the benchmark of 
+choice for `many <https://ieeexplore.ieee.org/abstract/document/8556738/>` 
+`machine <https://lib.dr.iastate.edu/cgi/viewcontent.cgi?article=1187&context=imse_conf>` 
+`learning <https://proceedings.neurips.cc/paper/1999/file/f3144cefe89a60d6a1afaf7859c5076b-Paper.pdf>`
+`papers <https://arxiv.org/search/?query=boston+housing&searchtype=all&source=header>`.
+As of scikit-learn version 1.2, the dataset has been removed.
```

Comment:
I think 1.0. 1.2 isn't out yet

Reply:
to be resolved in next commit

Reply:
I'm sorry, I was wrong! I should have checked yesterday but didn't quite get to it. Indeed it will be removed in 1.2, which is not yet out since they just released 1.0. So it's a very nice heads up for a future version that they're giving us. https://scikit-learn.org/stable/modules/generated/sklearn.datasets.load_boston.html?highlight=load_boston

Still, the original text wasn't quite right since it said "has been removed" rather than "will be removed".